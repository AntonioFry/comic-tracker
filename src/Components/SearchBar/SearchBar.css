.search-form {
  align-items: center;
  display: flex;
  height: 100px;
  justify-content: center;
  width: 100%;
}

.search-input {
  font-size: 1em;
  height: 30%;
  width: 25%;
}

.search-button {
  background-color: white;
  border-left: transparent;
  height: 30%;
}