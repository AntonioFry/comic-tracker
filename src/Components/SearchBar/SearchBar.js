import React, { Component } from 'react';

export class SearchBar extends Component {
  constructor() {
    super();
    this.state = {
      inputValue: '',
    }
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value })
  }

  render() {
    console.log(this.state.inputValue)
    return (
      <form>
        <input
          className='search-input'
          placeholder='Character Name...'
          value={this.state.inputValue}
          name='inputValue'
          onChange={(e) => this.handleChange(e)}
        />
        <button>Search</button>
      </form>
    )
  }
}