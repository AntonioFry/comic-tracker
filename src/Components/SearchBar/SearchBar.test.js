import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom';
import { SearchBar } from './SearchBar.js';
import { BrowserRouter } from 'react-router-dom';

describe('SearchBar', () => {

  it('Should match the snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <SearchBar />
      </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
});