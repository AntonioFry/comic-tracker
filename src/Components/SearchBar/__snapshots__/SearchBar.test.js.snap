exports[`SearchBar Should match the snapshot 1`] = `
<form
  className="search-form">
  <input
    className="search-input"
    name="inputValue"
    onChange={[Function]}
    placeholder="Character Name..."
    value="" />
  <button
    className="search-button"
    onClick={[Function]}>
    Search
  </button>
</form>
`;
